packages/@lwc/style-compiler/src/serialize.ts

```diff
@@ -86,7 +86,11 @@ export default function serialize(result: Result, config: Config): string {
     }
 
     // exports
-    buffer += `export default [${stylesheetList.join(', ')}];`;
+    if (stylesheetList.length) {
+        buffer += `export default [${stylesheetList.join(', ')}];`;
+    } else {
+        buffer += `export default undefined;`;
+    }
```

Comment:
Note this change is backwards-compatible with any shenanigans from custom Rollup plugins / Webpack / etc. If the CSS file exports `[]` then it just doesn't get the optimization.